test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env tarantool
```

Comment:
We should write all new new tests with luatest if possible. Is there anything in the test that can't be done with luatest that made you use TAP instead?

Reply:
I've started this patch not so long ago and it was legal to use TAP tests. Could I leave it as before? Of course for all new tests I will use luatest.

Reply:
I don't understand your reluctance. The test is short and should be pretty easy to convert to Luatest. Up to @Totktonada.

Anyway, the patch seems to have a bug: CTRL-C doesn't work after we exit a nested console:

```
tarantool> _ = require('console').listen(3301)
started
---
...

tarantool> require('console').connect('localhost:3301')
connected to localhost:3301
---
- true
...

localhost:3301> ^C
got signal 2 - Interrupt
```

Please fix and add a test case.

Reply:
> @locker Could you share the environment, where you got this bug, please? I can't reproduce it.

[ please reply in the thread ]

```sh
gh pr checkout 6633
cmake -S. -Bbuild/devug
make -Cbuild/debug -j16
build/debug/src/tarantool
```

If I use CTRL-D I exit from a nested console, however if I type CTRL-C, it hangs. Second CTRL-C terminates Tarantool.

```
Tarantool 2.10.0-beta2-111-ge9dee39a6082
type 'help' for interactive help
tarantool> _ = require('console').listen(3301)
started
---
...

tarantool> require('console').connect('localhost:3301')
connected to localhost:3301
---
- true
...

localhost:3301>                               << typed CTRL-D here, returned to parent console
tarantool> require('console').connect('localhost:3301')
connected to localhost:3301
---
- true
...

localhost:3301> ^C                            << typed CTRL-C here, console hangs
got signal 2 - Interrupt                      << typed CTRL-C again here, Tarantool exits
```

The expected behavior is CTRL-C shouldn't hang or terminate a nested console.

Reply:
> gh pr checkout 6633

What does this line mean? 
Anyway, after rebuild I can't reproduce this bug. For my environments (I use `Darwin Kernel Version 20.6.0 x86_64` and `Ubuntu kernel version 5.4.0-42-generic x86_64`) that works as expected:

```
tarantool> require('console').connect('localhost:3301')
connected to localhost:3301
---
- true
...

localhost:3301> ^C
---
...

localhost:3301> ^C
---
...

localhost:3301>
```

Reply:
> > gh pr checkout 6633
> 
> What does this line mean?

It checkouts the branch from this PR. It's a part of the GitHub CLI:

![image](https://user-images.githubusercontent.com/1114511/154083088-3632b91f-2422-47b7-847f-1ac8732bd540.png)

> Anyway, after rebuild I can't reproduce this bug.

Weird... Just tried to build your branch in the release mode, got the same results (hang).

I'm running Ubuntu 20.04:

```
vlad@esperanza:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
vlad@esperanza:~$ uname -a
Linux esperanza 5.13.0-28-generic #31~20.04.1-Ubuntu SMP Wed Jan 19 14:08:10 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

I think you should add a test case and see if it passes on CI.

Reply:
> I think you should add a test case and see if it passes on CI.

Seems really weird, but it passes on CI.

Reply:
I asked @alyapunov to check out this branch and he got a hang as well so it's not just me - looks like there is an issue.

Reply:
I confirm the problem.

We looked on it today with Slava and he reproduces the problem as well. The trick was that Slava tried to reproduce it by listening on one instance and connecting from another, not the same one.

Reply:
I will add a test for this case.

Reply:
Fixed.